neonvm/tools/vm-builder/files/resize-swap.sh

```diff
@@ -0,0 +1,33 @@
+#!/neonvm/bin/sh
+
+set -euo pipefail
+
+USAGE="USAGE: /neonvm/bin/resize-swap [ --once ] <NEW_SIZE_IN_BYTES>"
+
+if [ "$#" -eq 2 ]; then
+    if [ "$1" != '--once' ]; then
+        echo "expected first arg to be '--once'" >&2
+        echo "$USAGE" >&2
+        exit 1
+    fi
+    size="$2"
+    once=yes
+elif [ "$#" -eq 1 ]; then
+    size="$1"
+    once=no
+else
+    echo "expected 1 or 2 args, got $#" >&2
+    echo "$USAGE" >&2
+    exit 1
+fi
+
+if [ ! -f /neonvm/runtime/resize-swap.sh ]; then
+    echo 'not found: swap may not be enabled' >&2
+    exit 1
+fi
+
+/neonvm/bin/sh /neonvm/runtime/resize-swap.sh "$size"
+if [ "$once" = 'yes' ]; then
```

Comment:
Why the need for this?

Reply:
In short, so that if compute_ctl restarts it won't disable and re-enable swap — should help in edge cases if e.g. compute_ctl or postgres gets OOM killed, for example.

Reply:
I'd rather keep this logic on the `compute_ctl` side. From our point of view we can always do the swap resize, so it is up to the caller to decide whether to do it once or multiple times.

Reply:
Hm... The thing that makes that tricky is that compute_ctl _generally_ acts as if it's only run once, even though that's not the case. To avoid hitting this, we'd have to either (a) check whether the swap amount is already what we want (easier if we always `DisableSwapon`, so then it's just "is it zero"), or (b) create some file after resizing swap, to record that it's taken place.

Those both feel kind of hacky to me, but maybe they're better than supporting `--once` here? WDYT?

Reply:
I am ok with keeping it as-is, but I'd consider it tech debt if compute_ctl can't determine whether it has restarted or not. Feels like it might have implications in other parts.

But, on the other hand it is also ok if we provide indempotent execution environment for the compute_ctl, if we do it consistently.

Reply:
> but I'd consider it tech debt if compute_ctl can't determine whether it has restarted or not. Feels like it might have implications in other parts

Yeah 100% - this has caught people by surprise previously. It's kind of an open question what the desired behavior should be though... e.g. if the policy is "restart the VM when compute_ctl exits", then if compute_ctl gets OOM-killed that'll restart the entire VM

> But, on the other hand it is also ok if we provide indempotent execution environment for the compute_ctl, if we do it consistently.

Could you expand on that?

Reply:
> > but I'd consider it tech debt if compute_ctl can't determine whether it has restarted or not. Feels like it might have implications in other parts
> 
> Yeah 100% - this has caught people by surprise previously. It's kind of an open question what the desired behavior should be though... e.g. if the policy is "restart the VM when compute_ctl exits", then if compute_ctl gets OOM-killed that'll restart the entire VM

Makes sense, why don't we have that policy yet?

> 
> > But, on the other hand it is also ok if we provide indempotent execution environment for the compute_ctl, if we do it consistently.
> 
> Could you expand on that?

I meant explicitly acknowledging the ability of `compute_ctl` to restart and continue further execution as normal is also a valid contract. Seems redundant though.